### DISCUSSION TITLE
Problem during generating a single executable file.

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#820] [Q&A] [answered]
Hi Marco,
I just create a single exe by pyinstaller as the instructions in API Documentation.
But the exe file can't run due to the following errors:

![image](https://user-images.githubusercontent.com/116484589/221362770-b36616e6-656b-4eac-a5f1-c83a73c2168b.png)

For your convenience, all files all py files and spec file for pyinstaller are attached.
ps. the py file can run and has no errors.
[XTg_V1.zip](https://github.com/marcomusy/vedo/files/10831402/XTg_V1.zip)

**Comment by ZiguoAtGitHub** [comment, marked as answer]
Hi Marco,
I found the reason just a moment ago: The NoneType error of "vedo\__init__.py" is caused by  setting the console to False in the spec file.
Hope it be helpful for you to solve this problem.

**Reply by marcomusy** [reply]
Hi Ziguo, I cannot run your code (i'm on a mac and i get an error message which seems related to windows).
Please try to reproduce **exactly** this example, that seems to work fine:
https://github.com/marcomusy/welsh_embryo_stager#to-generate-a-standalone-executable

```
git clone https://github.com/marcomusy/welsh_embryo_stager.git
cd welsh_embryo_stager
...
```

**Reply by ZiguoAtGitHub** [reply]
Hi Marco, Thanks for your answer. It is a pity that you cannot reproduce the problems.
Here are some progresses: 
One good news: The ugly black dos window of console mode is killed off by lower version (v5.6.2) pyinstaller. :)
One bad news:
The exe can run well on my pc. But when I copy the execute file to another computer without python, vedo and other packages. I got these errors.

> [vedo.shapes.py:3874] WARNING: font name Wananti not found.
> Exception in Tkinter callback
> Traceback (most recent call last):
>   File "numpy\lib\npyio.py", line 441, in load
> EOFError: Ran out of input
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "vedo\shapes.py", line 3872, in _load_font
>   File "numpy\lib\npyio.py", line 443, in load
> _pickle.UnpicklingError: Failed to interpret file 'C:\\Users\\xuzig\\AppData\\Local\\Temp\\Wananti.npz' as a pickle
> 

> During handling of the above exception, another exception occurred:
>
>
> Traceback (most recent call last):
>   File "tkinter\__init__.py", line 1948, in __call__
>   File "XTg_New.py", line 185, in show_vse_dov_fig
>   File "vedo\addons.py", line 3681, in Axes
>   File "vedo\shapes.py", line 3983, in __init__
>   File "vedo\shapes.py", line 4152, in _get_text3d_poly
>   File "vedo\shapes.py", line 3882, in _get_font_letter
>   File "vedo\shapes.py", line 3875, in _load_font
> RuntimeError
> 
 I checked the path 'C:\\Users\\xuzig\\AppData\\Local\\Temp\\Wananti.npz' and did find the Wananti.npz file there.
And the problem is still there after some tries, eg. copy all font files into the same directory of exe file or put all font files into new fonts folder.  
It will be appreciated if you can give any suggestions.
Thanks again!

**Reply by marcomusy** [reply]
oh I see why it doesn't work!
you may try to copy Wananti.npz and Wananti.ttf to `vedo/fonts` directory **(where all the other fonts are)**.
then in your script add
```python
import vedo
...
settings.font_parameters["Wananti"]["islocal"] = True
```
then

```
pip uninstall vedo
pip install .
```

Wananti is public-domain so no problems with the license.
I hope this will work, let me know if not!

**Reply by ZiguoAtGitHub** [reply]
Hi Marco,
It works as you hope!
![033DB3FD](https://user-images.githubusercontent.com/116484589/221393692-b5f55275-ba4d-48c1-92e1-e501733d8952.png)
Thank you!!🤝🙏

**Reply by marcomusy** [reply]
great!
